lua/notebook-navigator/init.lua

```diff
@@ -258,6 +261,33 @@ M.setup = function(config)
   if (M.config.activate_hydra_keys ~= nil) and got_hydra then
     activate_hydra(M.config)
   end
+
+  local cell_group = vim.api.nvim_create_augroup("Cells", {clear=true})
+  if M.config.cell_folding then
+    local foldm = vim.o.foldmethod
+    vim.cmd[[set nofoldenable]]
+    vim.api.nvim_create_autocmd("BufEnter", {
+      pattern = {"*"},
+      group = cell_group,
+      callback = function(ev)
+        local ft = vim.api.nvim_get_option_value("filetype", {buf = ev.buf})
+        if M.config.cell_markers[ft] then
+          local marker = M.config.cell_markers[ft]
+          local vim_marker = string.gsub(marker, " ", "\\ ")
+          local expr = "set foldexpr=(getline(v:lnum)=~'^"..vim_marker.."')==0"
```

Comment:
This is going to overwrite whatever code folding a user has set. If you are only on folding cells that is fine but for example one thing that I do quite often after opening a file is zC to close all the folds to get a higher level view.

I would imagine that one approach to fix solve this is to wrap around the folding method currently in use so that it returns the fold level on top of the one that was set before.

Another issue is that this will changed the global fold method. Ideally one would rather use `setlocal` to only change it on the window displaying the buffer of interest.

Reply:
Really good point about `setlocal`. I've implemented that change.

On wrapping the current folding method. I think it could be done if the folding was set to `fold-indent`, `fold-expr`, or `fold-marker` but I'm not sure how to get an `expr` wrapping `fold-syntax` or `fold-diff`. Any ideas?

In any case, I've also added a warning for the time being that the code folding will replace previously defined `foldmethods` on buffers with valid cell markers.

Reply:
I would be happy with just having the cell folding working for a limited number of user set folding methods. Indent, expr and marker would be enough as they cover most of the situations.

Currently I'm wokring on wrapping around the output of the current foldexpr such that it produces the correct folding when cells are considered too. I will upload a branch later today or tomorrow morning sketching the idea.